h/static/scripts/controllers/search-bar-controller.js

```diff
@@ -137,16 +137,113 @@ class SearchBarController extends Controller {
       }
     };
 
-    var setupListenerKeys = event => {
+    /**
+     * Updates the hidden input field wtih the consolidated
+     * search terms from the lozenges.
+     */
+    var updateQueryParam = () => {
+      var searchTerms = '';
+      Array.from(this._lozenges.querySelectorAll('.js-lozenge__content'), function(loz) {
+        searchTerms = searchTerms + loz.textContent + ' ';
+      });
+      searchTerms = searchTerms + getTrimmedInputValue();
+      this._inputHidden.value = searchTerms;
+    };
+
+    /**
+     * Creates a lozenge and sets the content string to the
+     * content provided and executes the delete callback when
+     * the lozenge is deleted.
+     *
+     * @param {String} content The search term
+     */
+    var lozengify = content => {
+      var deleteCallback = () => {
+        updateQueryParam();
+        this.element.querySelector('form').submit();
+      };
+
+      new LozengeController(
+        this._lozenges,
+        {
+          content: content,
+          deleteCallback: deleteCallback,
+        }
+      );
+
+      // Clear the input after the lozenge is created and
+      // appended to the container element.
+      this._input.value = '';
+    };
+
+
+    /**
+     * Create lozenges for the search query terms already in
+     * the hidden input field on page load.
+     */
+    var lozengifyQueryParams = () => {
+      var queryTerms = SearchTextParser.getQueryTerms(this._inputHidden.value);
+      queryTerms.forEach(function(term) {
+        lozengify(term);
+      });
+    };
+
+    /**
+     * Setup listener keys with the handlers provided for each
+     * of them.
+     *
+     * @param {Event} The event.
+     * @param {Object} The function to execute when
+     * the event fires for each listener key.
+     */
+    var setupListenerKeys = (event, handlers) => {
+      var handler = handlers[event.keyCode];
+      if (handler) {
+        handler(event);
+      }
+    };
+
+    /**
+     * Setup the space key as the  listener for
+     * creating a lozenge.
+     *
+     * @param {Event} The event to listen for.
+     */
+    var setupLozengeListenerKeys = event => {
+      const SPACE_KEY_CODE = 32;
+
+      var handleSpaceKey = () => {
+        var word = getTrimmedInputValue();
+        if (SearchTextParser.shouldLozengify(word)) {
+          lozengify(word);
+        }
+      };
```

Comment:
Optionally this block (get text from the input field; make a lozenge out of that text; and then empty the input field) could move into a `lozengifyInputField()` function